packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -24,10 +24,11 @@ import 'globals.dart' as globals;
 import 'platform_plugins.dart';
 import 'project.dart';
 
-void _renderTemplateToFile(String template, dynamic context, String filePath) {
+void _renderTemplateToFile(String template, dynamic context, String filePath, {FileSystem fileSystem}){
   final String renderedTemplate = globals.templateRenderer
     .renderString(template, context, htmlEscapeValues: false);
-  final File file = globals.fs.file(filePath);
+  final FileSystem fs = fileSystem ?? globals.fs;
```

Comment:
what about just globals.fs?  it'd be more consistent with the use of globals in line 28

Reply:
this adds the possibility to over write fileSystem which makes test easier.